internal/common/physics/engine.go

```diff
@@ -51,16 +51,12 @@ func ComputeDistance(src utils.Coordinates, target utils.Coordinates) float64 {
 
 // This function is to be called from the server only
 func GenerateNewState(initialState utils.PhysicalState, force float64, orientation float64) utils.PhysicalState {
+	var finalState utils.PhysicalState = initialState
```

Comment:
What is the reason for having this? Why not just mutate initialState if you're going to go with mutating the input? The `inputState` argument will already be a copy since it's passed by value

Reply:
Yes, we could. That's just an alternative way to copy data from initial state to final state, in the last version, we created a new physicalState (finalState) and assign data to the final state. but sometimes it could miss sth as we are creating a brand new physical state like mass? so instead I just make final state equal to ininital state at start and then make change to those value we want to update.